Add build and ppm docs to core hacking section

Daeraxa opened the pull request:
This adds three new files into the index for "Hacking the Core"

`building-pulsar` contains the info for downloading the code, pre-reqs, dependencies and platform specific instructions

`using-ppm` has info on how to export the env variables to get ppm to work and an example of how you would use it

`help` has info for the project socials in case one needs assistance and a link to the pulsar repo for reporting issues with the build process

Feel free to comment on the structure, language used etc. If it needs significant rework then please indicate it and I'll convert to a draft PR whilst we come to a consensus about the structure or patterns we want to use in our docs.

confused-Techie commented on docs/docs/launch-manual/sections/core-hacking/using-ppm.md:
````diff
@@ -0,0 +1,45 @@
+@tab Windows
+
+```sh
+TODO
````
```suggestion
set ATOM_HOME=C:\Users\<user>\.pulsar
set APM_PATH=\ppm\bin\apm
```

confused-Techie reviewed (changes requested):
This overall looks fantastic, just added some suggestions for Windows instructions, as well as a slight syntax change.

Otherwise I'd just like to ensure someone is able to validate and test the Linux based instructions that are included.

kaosine commented on docs/docs/launch-manual/sections/core-hacking/help.md:
```diff
@@ -0,0 +1,9 @@
+## Having trouble?
+
+If you have any issues then please feel free to ask for help in the Pulsar
+[Discord server](https://discord.gg/gp9UStzsHk),
+[subreddit](https://www.reddit.com/r/pulsaredit/) or
+[GitHub Discussions](https://github.com/pulsar-edit/pulsar/discussions).
```
```suggestion
If you have any issues then please feel free to ask for help in the Pulsar
[Discord server](https://discord.gg/7aEbB9dGRT),
[subreddit](https://www.reddit.com/r/pulsaredit/) or
[GitHub Discussions](https://github.com/pulsar-edit/pulsar/discussions).
```

kaosine replied:
I had to add my own suggestion since that link @confused-Techie added for discord was off

kaosine replied:
Though honestly, I feel like these changes could go in a FAQ for the Documentation as a whole.

kaosine replied:
And by this I mean your troubleshooting.md. We'll need to make and section off for things like building anyways....

kaosine commented on docs/docs/launch-manual/sections/core-hacking/building-pulsar.md:
````diff
@@ -0,0 +1,118 @@
+
+@tab Windows
+
+```
+TODO
+```
````
```suggestion
First install [Visual Studio](https://visualstudio.microsoft.com/downloads/) from Microsoft.

::: info
 Install Node16 (using `nvm` - see above) and enable corepack (for `yarn`)
:::


 ```sh
nvm install 16
 corepack enable
 ```

Daeraxa replied:
@kaosine I think if we want the info blocks in the instructions we can redo the whole set with them after. For the moment just going to go with the code block style for consistency with the linux ones, we can revisit for polish later.

Daeraxa commented:
> This overall looks fantastic, just added some suggestions for Windows instructions, as well as a slight syntax change.
> 
> Otherwise I'd just like to ensure someone is able to validate and test the Linux based instructions that are included.

The OpenSUSE instructions are ones I added in the discord #build thread. The Fedora ones were created by @kaosine which I tested on my own Fedora laptop and it worked just fine.

kaosine commented on docs/docs/launch-manual/sections/core-hacking/help.md:
```diff
@@ -0,0 +1,9 @@
+## Having trouble?
+
+If you have any issues then please feel free to ask for help from the Pulsar Team:
+- [Discord server](https://discord.gg/gp9UStzsHk)
+- [subreddit](https://www.reddit.com/r/pulsaredit/)
+- [GitHub Discussions](https://github.com/pulsar-edit/pulsar/discussions)
+
+If you think you have found a bug in the process then please create a
+[new issue](https://github.com/pulsar-edit/pulsar/issues).
```
Let's move this to a general FAQ section. This is the only other change I'd make. Then we could include specific lines as needed in these sections, but have it generally accessible from a main link.

```
@include(filename{start-end})
```
(from the documentation on this, we'll have to make sure to update the readme with this kind of information as well. But I think this will make things a bit simpler especially once something like this is.

There's probably even some other configs we can add since I added the theme last night:
https://vuepress-theme-hope.github.io/v2/guide/markdown/include.html#advanced

Daeraxa replied:
See new latest commits and comments

kaosine commented on docs/docs/launch-manual/sections/core-hacking/help.md:
```diff
@@ -0,0 +1,9 @@
+## Having trouble?
+
+If you have any issues then please feel free to ask for help from the Pulsar Team:
+- [Discord server](https://discord.gg/gp9UStzsHk)
```
```suggestion
- [Discord server](https://discord.gg/7aEbB9dGRT)
```
This is also using the wrong link

Daeraxa commented:
Latest update adds a new FAQ section which contains the "get-help" file which has been moved to those sections from the hacking the core section.
The hacking the core .md files have been moved inside a "sections" folder for more consistent organisation

kaosine reviewed (approved):
Approving these for now and merging them. We can make further edits and updates after they're merged in.
